webservices/resources/reports.py

```diff
@@ -105,11 +134,18 @@ def build_query(self, committee_id=None, committee_type=None, **kwargs):
         if kwargs.get('is_amended') is not None:
             query = query.filter(reports_class.is_amended == kwargs['is_amended'])
 
+        query = filters.filter_range(query, kwargs, self.filter_range_fields)
         return query, reports_class, reports_schema
 
     def _resolve_committee_type(self, committee_id=None, committee_type=None, **kwargs):
-        if committee_id is not None:
-            query = models.CommitteeHistory.query.filter_by(committee_id=committee_id)
+        #if kwargs.get('committee_id') is not None and len(kwargs.get('committee_id')) > 0:
+        if committee_id is not None or kwargs.get('committee_ids'):
+            comm_id = (
+                committee_id
+                if committee_id is not None
+                else kwargs.get('committee_ids')[0]
```

Comment:
Currently the query parameter for `auth_committee_candidate_id` is a list.  Which if you think of it runs up to some of the same problems with multiple committees, i.e. multiple candidates with the possibility of being different types.  So should I make it truly `candidate_id` and just call it a day.  

With features like this the problems that a user could create through even somewhat sensical queries runs up into exponential numbers of possible combinations.  So I think for any given feature you can end game it to death, but, I think your points are valid.

With that being said I'll change `auth_committee_candidate_id` to `candidate_id`, and parse out it's type based on the first character.